# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/audioipc/src/async_msg.rs
# Commit: 634412ce2943
# Full Hash: 634412ce29430557eec6d0c0313f1908e61ad13d
# Author: Matthew Gregan <kinetik@flim.org>
# Date: 2021-08-11 03:46:10
# Regressor Bug: 1724141
# File Overlap Count: 6
# Description:
#   Bug 1724141 - Update audioipc to 8bb1a227.  r=chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D121826
# ==============================================================================

diff -r 722284bed619 -r 634412ce2943 third_party/rust/audioipc/src/async_msg.rs
--- a/third_party/rust/audioipc/src/async_msg.rs	Tue Aug 10 20:38:51 2021 +0000
+++ b/third_party/rust/audioipc/src/async_msg.rs	Tue Aug 10 20:52:03 2021 +0000
@@ -8,9 +8,7 @@
 #[cfg(unix)]
 use crate::msg::{RecvMsg, SendMsg};
 use bytes::{Buf, BufMut};
-#[cfg(unix)]
-use futures::Async;
-use futures::Poll;
+use futures::{Async, Poll};
 #[cfg(unix)]
 use iovec::IoVec;
 #[cfg(unix)]
@@ -166,3 +164,31 @@
         }
     }
 }
+
+// AsyncRecvMsg/AsyncSendMsg are implemented for convenience on Windows and don't use the _cmsg parameter.
+#[cfg(windows)]
+impl AsyncRecvMsg for super::AsyncMessageStream {
+    fn recv_msg_buf<B>(&mut self, buf: &mut B, _cmsg: &mut B) -> Poll<(usize, i32), io::Error>
+    where
+        B: BufMut,
+    {
+        // _cmsg unused on Windows.  Pass through to read_buf.
+        match <super::AsyncMessageStream>::read_buf(self, buf) {
+            Ok(Async::Ready(n)) => Ok(Async::Ready((n, 0))),
+            Ok(Async::NotReady) => Ok(Async::NotReady),
+            Err(e) => Err(e),
+        }
+    }
+}
+
+#[cfg(windows)]
+impl AsyncSendMsg for super::AsyncMessageStream {
+    fn send_msg_buf<B, C>(&mut self, buf: &mut B, _cmsg: &C) -> Poll<usize, io::Error>
+    where
+        B: Buf,
+        C: Buf,
+    {
+        // _cmsg unused on Windows.  Pass through to write_buf.
+        <super::AsyncMessageStream>::write_buf(self, buf)
+    }
+}